src/core/public/http/http_service.test.ts

```diff
@@ -505,6 +559,112 @@ describe('interception', () => {
       'Response 3',
     ]);
   });
+
+  it('should not have access to request via response error interceptor when request error', async () => {
```

Comment:
I think these tests would be much easier to read if they were organized by `describe` blocks to structure the different permutations that need to be tested. Right now it takes a decent amount of thinking to understand if all the cases are covered.